wiki/History_of_osu!/April_Fools/en.md

```diff
@@ -201,94 +201,104 @@ For the April Fools' day of 2019, there was approximately a 1 in 20 chance of he
-
-[r]: #references
+[^lemontree-reddit]: [Reddit post by u/5522Luca in r/osugame (2017-04-10) "Reminder the Osu! April Fools 2009? This beatmap was ranked."](https://www.reddit.com/r/osugame/comments/64it62/reminder_the_osu_april_fools_2009_this_beatmap/)
+[^lemontree-post-machol30]: [Forum post by machol30 (2009-04-03) in "Best of No.1 Hits - Lemon Tree"](https://osu.ppy.sh/community/forums/posts/106774)
+[^lemontree-post-peppy]: [Forum post by peppy (2009-04-01) in "Best of No.1 Hits - Lemon Tree"](https://osu.ppy.sh/community/forums/posts/105679)
+[^lemontree-post-millhioref]: [Beatmap by MillhioreF (2012-08-24) "Best of No.1 Hits - Lemon Tree"](https://osu.ppy.sh/beatmapsets/57878#osu/174267)
+
+[^touhousu-ontheweb]: ["osu.ppy.sh - Changed osu! to touhousu! throughout the website as well as the game." on April Fools' Day On The Web](http://aprilfoolsdayontheweb.com/joke/8120/?size=1)
+[^touhousu-osudev-2021-01-27]: [Discord message by Nivalyx in #osu-wiki in osu!dev (2021-01-27)](https://discord.com/channels/188630481301012481/218677502141399041/804215894762848296)
+[^touhousu-forums]: [Forum thread by rcmero (2010-04-01) "touhousu! - April Fools joke? [Resolved]"](https://osu.ppy.sh/community/forums/topics/27612)
+[^touhousu-forums-2]: [Forum thread by rulingvenus (2010-04-01) "Naked Marisa????"](https://osu.ppy.sh/community/forums/topics/27531)
+
+[^nightcore-wiki]: ["Nightcore (mod)" on osu! wiki](/wiki/Game_modifier/Nightcore)
+[^nightcore-yt]: [YouTube video by Nyaruko (2011-03-31) "When osu! tries to do April Fools"](https://www.youtube.com/watch?v=qD5ep6Fykao)
+[^nightcore-frontpage]: ["osu! — rhythm is just a click away" (2011-04-01) on Wayback Machine](https://web.archive.org/web/20110401175252/http://osu.ppy.sh/)
+
+[^fl-forums]: [Forum post by Melty Bagle (2012-03-31) in "[Archived] 'Flashlight mod' on the site...?"](https://osu.ppy.sh/community/forums/posts/1430529)
+[^fl-osudev-2021-01-28]: [Discord message by TicClick in #osu-wiki in osu!dev (2021-01-28)](https://discord.com/channels/188630481301012481/218677502141399041/804260860331687946)
+[^fl-ontheweb]: ["osu.ppy.sh — 'Flashlight' mode on beatmap search page" on April Fools' Day On The Web](http://aprilfoolsdayontheweb.com/joke/11484/?size=1)
+[^fl-forums-2]: [Forum thread by ----- (2012-03-31) "[Archived] 'flashlight mod' on the site...?"](https://osu.ppy.sh/community/forums/topics/79076)
+[^fl-forums-3]: [Forum post by peppy (2012-04-01) in "[Archived] 'flashlight mod' on the site...?"](https://osu.ppy.sh/community/forums/posts/1433063)
+[^fl-forums-4]: [Forum thread by kreph (2012-03-31) "[Archived] Flashlight bugs the website for some browsers"](https://osu.ppy.sh/community/forums/topics/79077)
+[^fl-osudev-2021-01-29]: [Discord message by spaceman_atlas in #osu-wiki in osu!dev (2021-01-29)](https://discord.com/channels/188630481301012481/218677502141399041/804814051209117696)
```

Comment:
this just has the gifs that are already present [on the page](https://osu.ppy.sh/wiki/en/History_of_osu%21/April_Fools#flashlight/hidden-mod-on-website)

Reply:
I think that is still useful, in the sense of explaining where the gifs come from

Reply:
hmm i guess it's fine since there's a similar instance with [the one about the touhosu image](https://github.com/ppy/osu-wiki/pull/8093/files#diff-4678189be705fdff9210edf5dcff52ba9df35eaa216e1299fa50ec53060e718fR222)